rust/src/session.rs

```diff
@@ -67,34 +67,24 @@ type ArcAny = Arc<dyn Any + Send + Sync>;
 /// The cache is keyed by the file path and the type of metadata.
 #[derive(Clone)]
 pub struct FileMetadataCache {
-    /// The maximum number of metadata to cache.
-    capacity: usize,
-
-    cache: Arc<Mutex<LruCache<(Path, TypeId), ArcAny>>>,
+    cache: Arc<Cache<(Path, TypeId), ArcAny>>,
 }
 
 impl FileMetadataCache {
-    pub(crate) fn new(capacity: usize) -> Self {
```

Comment:
Did we remove the `pub(crate)` for the sake of benchmarking? Should we add them back?

Reply:
I've added these back in.  Thanks for catching.